WooCommerce/Classes/CardPresentPayments/CardPaymentsAdaptor.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+import Yosemite
+import class WooFoundation.CurrencyFormatter
+
+class CardPresentPaymentsAdaptor {
```

Comment:
💬  Payments newbie questions:

- Is this meant to be a singleton in the app for any card-present payments-related actions and states, including onboarding, reader connection, and payment collection from any entry point? While looking at the reader connection code, the Menu Payments setup seems different from the Orders tab. This adapter solves the problem with one single implementation that maintains various states, with `pos` being a new entry point.
- It looks like the UI/UX for the payment collection flow is fixed? It's fine if we plan to reuse the UI/UX for MVP, or just the first version of the MVP.
- For reuse in the current POS framework, do you think it's possible to have a protocol for this in Yosemite (given that we don't have plans to create new frameworks) so that the singleton uses the interface? It might help with unit testing as well.

Reply:
> Is this meant to be a singleton in the app for any card-present payments-related actions and states, including onboarding, reader connection, and payment collection from any entry point? [...] This adapter solves the problem with one single implementation that maintains various states, with pos being a new entry point.

Yes, exactly.

We may _also_ want to add a separate `connection` function from the same entry point, but it didn't seem essential to this experiment.

In the app right now, the vast majority of people connect the reader during payment.

> While looking at the reader connection code, the Menu Payments setup seems different from the Orders tab.

Yeah, the menu setup screen is a bit different, but it does use the connection controllers. It would be better if we had a single source of truth for these controllers, which we could add via this adaptor in the MVP or beyond, but it's not essential.

> For reuse in the current POS framework, do you think it makes sense to have a protocol for this in Yosemite (given that we don't have plans to create new frameworks) so that the singleton uses the interface? It might help with unit testing as well.

I've written this in the `App` framework, initially thinking about the situation of POS being in the `App` framework and removing the `POS` framework.

If we do continue with the `POS` framework, we have to make some choices if we use this approach...

We certainly could move the code that this depends on to Yosemite, but the singleton itself would need to live there too; using a shared interface wouldn't get us the code reuse benefits of the adaptor.

Alternatively, we could copy all the card reader code, and this adaptor, from the `App` framework to the `POS` framework, and build it out from there. That might let us improve on a "fork" (sort of) of the card reader code for the POS, which we could then merge back to the app target, or move down to Yosemite for sharing.